src/schemas.jl

```diff
@@ -375,12 +413,13 @@ macro schema(schema_name, schema_prefix)
     schema_prefix isa Symbol || return :(throw(ArgumentError(string("`Prefix` provided to `@schema` is not a valid type name: ", $(Base.Meta.quot(schema_prefix))))))
     return quote
         # This approach provides some safety against accidentally replacing another module's schema's name,
-        # without making it annoying to reload code/modules in an interactice development context.
+        # without making it annoying to reload code/modules in an interactive development context.
         m = $Legolas._schema_declared_in_module(Val(Symbol($schema_name)))
         if m isa Module && string(m) != string(@__MODULE__)
             throw(ArgumentError(string("A schema with this name was already declared by a different module: ", m)))
         else
             $Legolas._schema_declared_in_module(::Val{Symbol($schema_name)}) = @__MODULE__
+            $Legolas.schema_provider(::Val{Symbol($schema_name)}) = $Legolas.defining_package(@__MODULE__)
```

Comment:
oh i see, this a parallel to `_schema_declared_in_module`.  doesn't that do what you want here?

Reply:
I actually think it might not: even if the _schema_ is defined, the _version_ might not, which is just as bad as far as dealing with the arrow metadata is concerned...

Reply:
yeah, I wanted the defining package as a symbol, which is slightly different than `_schema_declared_in_module` but could be derived from it. But I agree, probably I should move this to the `@version` macro and use the full SV